### PR TITLE
C#/Java: Re-factor provenance related predicates.

### PR DESCRIPTION
**Pull request by michaelnebel**
In this PR we re-factor some of the predicates that is related to the provenance column in the MaD format. This is to avoid converting from string -> bool -> string in different layers of the code.

### CODE REVIEW

**Comment by michaelnebel** [comment]
DCA for C# and Go looks good.
Will re-rerun for Java after the merge conflicts have been fixed.

**Review by atorralba — APPROVED** [review]
Java LGTM

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll [hunk 1]
```diff
@@ -241,31 +241,39 @@ module Public {
     }
 
     /**
-     * Holds if the summary is auto generated and not manually generated.
+     * Holds if the summaries that apply to `this` are auto generated and not manually generated.
```
Comment (tamasvajk) [inline csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll:244]: Minor: Does "manually generated" make sense? `manual` models are not generated, are they.

**Review by tamasvajk — DISMISSED** [review]
C# 👍

File: csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll (hunk 1 above)
Comment (michaelnebel) [reply csharp/ql/lib/semmle/code/csharp/dataflow/internal/FlowSummaryImpl.qll:244]: Ah yes, I will change the wording to "manually created".

**Comment by michaelnebel** [comment]
@tamasvajk : Could you take a quick look. The only change since last time is the change in the comment.

**Review by tamasvajk — APPROVED** [review]

**Pull request merged**